src/components/project/GroupCheckboxListComp.vue

```diff
@@ -0,0 +1,66 @@
+<script setup lang="ts">
+import { computed, onMounted, ref, watch } from 'vue'
+import { useRoute } from 'vue-router'
+import { useProjectMemberGroupsStore } from '@/stores/ProjectMemberGroupsStore'
+
+const route = useRoute()
+const projectId = route.params.id
+
+type Group = {
+  group_id: number
+  group_name: string
+  description: string
+}
+
+const props = defineProps<{
+  value?: number[]
+  name: string
+}>()
+
+const projectMemberGroupsStore = useProjectMemberGroupsStore()
+const groups = computed((): Group[] => projectMemberGroupsStore.groups)
```

Comment:
I wanted to use it to constantly check for "groups" changing in the store so that it can then update the value in the component but then after reading the documentation for pinia I realized there was no need for that since the store is reactive and would update on its own. So I removed that line and in the template I use "projectMemberGroupsStore.groups".